From b82814f289ed4a99588e831e5fb5ff2866d69e2a Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 27 May 2016 12:45:41 -0500
Subject: [PATCH 5/7] Revert "mmc: host: omap_hsmmc: use generic_cmd6_time to
 program timeout value for CMD6"

This reverts commit eed2d405adf1a03051c65bb0dbf2f634f9590d91.

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mmc/host/omap_hsmmc.c | 20 ++++++--------------
 1 file changed, 6 insertions(+), 14 deletions(-)

diff --git a/drivers/mmc/host/omap_hsmmc.c b/drivers/mmc/host/omap_hsmmc.c
index 78caf3b..93019c9 100644
--- a/drivers/mmc/host/omap_hsmmc.c
+++ b/drivers/mmc/host/omap_hsmmc.c
@@ -1631,24 +1631,16 @@ static int
 omap_hsmmc_prepare_data(struct omap_hsmmc_host *host, struct mmc_request *req)
 {
 	int ret;
-	unsigned int timeout;
-
 	host->data = req->data;
 
 	if (req->data == NULL) {
 		OMAP_HSMMC_WRITE(host->base, BLK, 0);
-		if (req->cmd->flags & MMC_RSP_BUSY) {
-			timeout = req->cmd->busy_timeout * NSEC_PER_MSEC;
-
-			/*
-			 * Set an arbitrary 100ms data timeout for commands with
-			 * busy signal and no indication of busy_timeout.
-			 */
-			if (!timeout)
-				timeout = 100000000U;
-
-			set_data_timeout(host, timeout, 0);
-		}
+		/*
+		 * Set an arbitrary 100ms data timeout for commands with
+		 * busy signal.
+		 */
+		if (req->cmd->flags & MMC_RSP_BUSY)
+			set_data_timeout(host, 100000000U, 0);
 		return 0;
 	}
 
-- 
2.8.1

